scheduler/src/cook/tools.clj

```diff
@@ -916,27 +916,50 @@
        first
        field))
 
-(defn filter-based-on-quota
+(defn get-quota-for-pool
+  "Given a pool name, determine the quota for that pool."
+  [quotas effective-pool-name]
+  (match-based-on-pool-name quotas effective-pool-name :quota))
+
+(defn filter-based-on-user-quota
   "Lazily filters jobs for which the sum of running jobs and jobs earlier in the queue exceeds one of the constraints,
    max-jobs, max-cpus or max-mem"
   [user->quota user->usage queue]
   (letfn [(filter-with-quota [user->usage job]
             (let [user (:job/user job)
                   job-usage (job->usage job)
                   user->usage' (update-in user->usage [user] #(merge-with + job-usage %))]
-              (log/debug "Quota check" {:user user
-                                        :usage (get user->usage' user)
-                                        :quota (user->quota user)})
+              (log/debug "User quota check" {:user user
+                                             :usage (get user->usage' user)
+                                             :quota (user->quota user)})
               [user->usage' (below-quota? (user->quota user) (get user->usage' user))]))]
     (filter-sequential filter-with-quota user->usage queue)))
 
+(defn filter-based-on-pool-quota
+  "Lazily filters jobs for which the sum of running jobs and jobs earlier in the queue exceeds one of the constraints,
+   max-jobs, max-cpus or max-mem"
+  [quota usage queue]
+  (log/debug "Pool quota and usage:" {:quota quota :usage usage})
+  (if (nil? quota)
+    queue
+    (letfn [(filter-with-quota [usage job]
+              (let [job-usage (job->usage job)
+                    usage' (merge-with + job-usage usage)]
+                (log/debug "Pool quota check" {:usage usage'
+                                               :quota quota})
+                [usage' (below-quota? quota usage')]))]
+      (filter-sequential filter-with-quota usage queue))))
+
 (defn filter-pending-jobs-for-autoscaling
   "Lazily filters jobs to those that should be considered
   for autoscaling purposes. Note that this is used in two
   places: the /queue endpoint (Mesos only) and as an
   argument to scheduler/trigger-autoscaling! (k8s only)."
-  [user->quota user->usage queue]
-  (filter-based-on-quota user->quota user->usage queue))
+  [user->quota user->usage pool-quota queue]
+  (let [pool-usage (reduce (partial merge-with +) (vals user->usage))]
```

Comment:
Same comment as above about it being hard to tell what the shape of `pool-quota` and `pool-usage` are.

Reply:
Done.